src/PrestaShopBundle/Resources/config/services/core/grid/doctrine_query_builder.yml

```diff
@@ -66,7 +66,7 @@ services:
         arguments:
             - '@prestashop.core.query.doctrine_search_criteria_applicator'
             - "@=service('prestashop.adapter.legacy.context').getContext().language.id"
-            - '@=service("prestashop.adapter.shop.context").getContextListShopID()'
+            - '@=service("prestashop.adapter.shop.context").getContextListShopID(parameter("const_shop_share_customer"))'
```

Comment:
To sum it up:
It was implemented in another way in legacy customers listing
but ... it seems I can fix the bug passing a single param to what was used before
so: I was reused a legacy function, but different from the one that was used before

Reply:
Really cool if we can fix it like that! :+1:

Reply:
The only question is: are we sure that `prestashop.core.grid.query_builder.customer` is only used in this context and/or always needs to use the group sharing?

Reply:
@jolelievre ah you're right ... I think I "forced" the "shared data" behavior although this is an option that merchant can choose not to use

Reply:
Mmm, so does it mean you should use the value from a `Configuration::get` (if it exists) instead of the const value?

Reply:
I'm going to need to re-implement legacy way to work
which means this `if/else`
https://github.com/PrestaShop/PrestaShop/blob/develop/classes/shop/Shop.php#L1077

to make it short
```
if ($group['share_customer']) {
    $ids = Shop::getContextListShopID($share));
} else {
    $ids = Shop::getContextListShopID())
}
```

and I need to do this at configuration/constructor level 🤔 

I'm afraid a BC break will be needed 😭

Reply:
You mean a BC because you will have to change the constructor of `CustomerQueryBuilder`

Maybe you could instead create a new service `ShopRepository` with a `getContextListId()` method that would manage this if/else case Then you can simply change the service definition and there is no BC (all the naming can of course be improved, it's a quick proposition)

Reply:
Idea is good 👍